javatests/arcs/core/analysis/ExpressionClaimDeducerTest.kt

```diff
@@ -0,0 +1,338 @@
+
+    assertThat(actual).isEqualTo(Deduction.Path())
+  }
+
+  @Test
+  fun literal_boolean() {
+    val expr = PaxelParser.parse("false")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(Deduction.Path())
+  }
+
+  @Test
+  fun literal_null() {
+    val expr = PaxelParser.parse("null")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(Deduction.Path())
+  }
+
+  @Test
+  fun single_variable() {
+    val expr = PaxelParser.parse("x")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(Deduction.Path("x"))
+  }
+
+  @Test
+  fun variable_field_access() {
+    val expr = PaxelParser.parse("x.foo")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Path("x", "foo")
+    )
+  }
+
+  @Test
+  fun variable_field_access_nested() {
+    val expr = PaxelParser.parse("x.foo.bar")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Path("x", "foo", "bar")
+    )
+  }
+
+  @Test
+  fun variable_field_access_binop() {
+    val expr = PaxelParser.parse("x.foo.bar + y.foo.bar.baz + z.baz.bar")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Paths(
+        listOf("x", "foo", "bar"),
+        listOf("y", "foo", "bar", "baz"),
+        listOf("z", "baz", "bar")
+      )
+    )
+  }
+
+  @Test
+  fun variable_field_access_binop_literals() {
+    val expr = PaxelParser.parse("x.foo.bar + y.foo.bar.baz + z.baz.bar + 10 + 20")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Paths(
+        listOf("x", "foo", "bar"),
+        listOf("y", "foo", "bar", "baz"),
+        listOf("z", "baz", "bar"),
+        emptyList(),
+        emptyList()
+      )
+    )
+  }
+
+  @Test
+  fun new() {
+    val expr = PaxelParser.parse("new Object {foo: x, bar: y}")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "foo" to Deduction.Path("x"),
+        "bar" to Deduction.Path("y")
+      )
+    )
+  }
+
+  @Test
+  fun new_field_access() {
+    val expr = PaxelParser.parse("new Object {foo: input.foo, bar: input.foo.bar}")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Scope(
+        "foo" to Deduction.Path("input", "foo"),
+        "bar" to Deduction.Path("input", "foo", "bar")
+      )
+    )
+  }
+
+  @Test
+  fun sum_variables() {
+    val expr = PaxelParser.parse("x + y")
+
+    val actual = expr.accept(ExpressionClaimDeducer(), Unit)
+
+    assertThat(actual).isEqualTo(
+      Deduction.Paths(
+        listOf("x"),
+        listOf("y")
+      )
+    )
+  }
+
+  @Test
+  fun sum_variables_field_access() {
```

Comment:
looks the same as variable_field_access_binop

Reply:
removed.